stories/dialogs.js

```diff
@@ -21,81 +22,58 @@ storiesOf('Dialogs', module)
-            <Heading3>Dialog title</Heading3>
-          </Banner>
-          <Box padding={4}>
-            <TextBody>Here you can add arbitrary content.</TextBody>
-          </Box>
-          <ButtonGroup justifyContent="flex-end" padding={4}>
-            <Button label="Cancel" />
-            <Button label="Confirm" level="primary" />
-          </ButtonGroup>
-        </Dialog>
-      </State>
-    </Box>
-  ))
-  .add('succes style', () => (
-    <Box>
-      <Button onClick={handleActiveToggle} label="Open a dialog" />
-      <State store={store}>
-        <Dialog active={false} onEscKeyDown={handleActiveToggle} onOverlayClick={handleActiveToggle}>
-          <Banner color="mint" fullWidth icon={<IconCheckmarkMediumOutline />} onClose={handleActiveToggle}>
-            <Heading3>Succes: Dialog title</Heading3>
-          </Banner>
-          <Box padding={4}>
-            <TextBody>Here you can add arbitrary content.</TextBody>
-          </Box>
-          <ButtonGroup justifyContent="flex-end" padding={4}>
-            <Button label="Cancel" />
-            <Button label="Confirm" level="primary" />
-          </ButtonGroup>
-        </Dialog>
-      </State>
-    </Box>
-  ))
-  .add('warning style', () => (
-    <Box>
-      <Button onClick={handleActiveToggle} label="Open a dialog" />
-      <State store={store}>
-        <Dialog active={false} onEscKeyDown={handleActiveToggle} onOverlayClick={handleActiveToggle}>
-          <Banner color="gold" fullWidth icon={<IconWarningBadgedMediumOutline />} onClose={handleActiveToggle}>
-            <Heading3>Warning: Dialog title</Heading3>
-          </Banner>
-          <Box padding={4}>
+  .add('DialogBase', () => {
+    return (
+      <Box>
+        <Button onClick={handleActiveToggle} label="Open a dialog base" />
+        <State store={store}>
+          <DialogBase
+            backdrop={select('backdrop', ['dark'], 'dark')}
+            onEscKeyDown={handleActiveToggle}
+            onOverlayClick={handleActiveToggle}
+            size={select('size', sizes, 'medium')}
+          >
+            <Box padding={4}>
+              <TextBody>Here you can add arbitrary content.</TextBody>
+            </Box>
+          </DialogBase>
+        </State>
+      </Box>
+    );
+  })
+  .add('Dialog', () => {
+    const passHeaderIcon = boolean('pass a headerIcon', false);
+
+    return (
+      <Box>
+        <Button onClick={handleActiveToggle} label="Open a dialog" />
+        <State store={store}>
+          <Dialog
+            headerColor={select('headerColor', COLORS, 'neutral')}
+            headerIcon={passHeaderIcon && <IconWarningBadgedMediumOutline />}
+            headingLevel={number('headingLevel', 3, { min: 2, max: 3 })}
+            onCloseClick={handleActiveToggle}
+            primaryAction={{
+              label: text('primaryAction.label', 'Confirm'),
+              onMouseUp: () => console.log('primaryAction.onMouseUp'),
```

Comment:
Q: any reason why you're using `onMouseUp` and not just `onClick`?

Reply:
None in particular. You could actually pass down any prop or attribute that you can pass to a `Button`.

But it makes more sense to use `onClick` in this case so I'm going to change it.